onnxruntime/core/optimizer/pad_fusion.cc

```diff
@@ -115,18 +146,22 @@ Status PadFusion::Apply(Graph& graph, Node& pad_node, RewriteRuleEffect& rule_ef
   }
 
   Node& child_node = *graph.GetNode(pad_node.OutputNodesBegin()->Index());
-  auto child_pads = child_node.GetMutableAttributes()["pads"].mutable_ints();
-  uint32_t child_pads_size = static_cast<uint32_t>(child_pads->size());
-
-  for (uint32_t pads_index = 2, child_index = 0; pads_index < pads_size / 2; pads_index++, child_index++) {
-    child_pads->Set(child_index, child_pads->Get(child_index) + pads_values[pads_index]);
-    uint32_t mirrored_child_index = child_index + (child_pads_size / 2);
-    uint32_t mirrored_pad_index = pads_index + (pads_size / 2);
-    child_pads->Set(mirrored_child_index, child_pads->Get(mirrored_child_index) + pads_values[mirrored_pad_index]);
+  if (child_node.OpType() == "Cast") {
+    // We don't need to cast the pad_constant_value because this fusion requires that constant_pad_value
+    // to be zero. See PadFusion::SatisfyCondition for details.
+    Update_Pad_Attribute(*graph.GetNode(child_node.OutputNodesBegin()->Index()), pads_values, pads_size);
+  } else {
+    Update_Pad_Attribute(child_node, pads_values, pads_size);
   }
 
   graph_utils::RemoveNodeOutputEdges(graph, pad_node);
   graph_utils::ReplaceNodeInput(child_node, 0, *pad_node.MutableInputDefs()[0]);
+  
+  // Un-pad the output shape of Cast node
+  if (child_node.OpType() == "Cast") {
```

Comment:
Q: Can `ReplaceNodeInput` potentially invalidate the reference to `child_node`? (unsure if it reallocates any memory or not)

Reply:
As per the body of the function, it won't.